src/main/java/io/cryostat/net/web/http/api/v2/TargetsPostHandler.java

```diff
@@ -156,10 +164,42 @@ public IntermediateResponse<ServiceRef> handle(RequestParameters params) throws
                     throw new ApiException(400, "Duplicate connectUrl");
                 }
             }
-            Map<AnnotationKey, String> cryostatAnnotations = new HashMap<>();
 
-            String jvmId = jvmIdHelper.getJvmId(uri.toString());
+            MultiMap queries = params.getQueryParams();
+            boolean dryRun =
+                    StringUtils.isNotBlank(queries.get("dryrun"))
+                            && Boolean.valueOf(queries.get("dryrun"));
+
+            String username = attrs.get("username");
+            String password = attrs.get("password");
+            Optional<Credentials> credentials =
+                    StringUtils.isBlank(username) || StringUtils.isBlank(password)
+                            ? Optional.empty()
+                            : Optional.of(new Credentials(username, password));
+
+            if (!dryRun && credentials.isPresent()) {
+                String matchExpression = String.format("target.connectUrl == '%s'", connectUrl);
```

Comment:
https://github.com/cryostatio/cryostat/blob/bcf8181fd0ea1ae73ee4dad582bbc9bb948491d8/src/main/java/io/cryostat/configuration/CredentialsManager.java#L157

Reply:
Right! Updated. Any background on why this method is deprecated?

Reply:
It's related to the change from target-specific to matchExpression-based JMX credentials. Before your usage of it here it was only used by old, deprecated 2.1 credentials API endpoints, so that those endpoints could still exist but feed into the new matchExpression system underneath.

You can un-deprecate the method since this is another valid usecase for it.

Reply:
Make sense! Thanks for explaining! I undid the deprecation now.